website/content/guides/1.7-template-upgrade.mdx

```diff
@@ -0,0 +1,154 @@
+| Not Compatible                   | Manual and `packer init` Installation  | Manual and `packer init`  Installation | v1.7.0           |
+
+### How to upgrade my template
+
+Let's use the following template as an example:
+
+```hcl
+# file: example.pkr.hcl
+
+source "null" "basic-example" {
+  communicator = "none"
+}
+
+build {
+  sources = ["sources.null.basic-example"]
+
+  provisioner "comment" {
+    comment = "Hello from comment plugin"
+    ui = true
+    bubble_text =  true
+  }
+}
+```
+
+This template uses the `packer-provisioner-comment` plugin and, until Packer v1.7.0, the binaries were installed manually.
+
+From Packer v1.7.0, the plugin has been upgraded by its maintainers and now is called `packer-plugin-comment` and the
+latest version is v0.2.23. Knowing that, our example template can be upgraded.
+
+```hcl
+# file: example.pkr.hcl
+
+packer {
+    required_plugins {
+        comment = {
+            version = ">=v0.2.23"
+            source = "sylviamoss/comment"
+        }
+    }
+}
+
+source "null" "basic-example" {
+  communicator = "none"
+}
+
+build {
+  sources = ["sources.null.basic-example"]
+
+  provisioner "comment" {
+    comment = "Hello from comment plugin"
+    ui = true
+    bubble_text =  true
+  }
+}
+```
+
+The upgraded template is now telling Packer that it requires a plugin from the source `sylviamoss/comment`
+of the version equals or greater `v0.2.23`. It also tells that the plugin should be referred in the rest of the template as `comment`.
+
+Here it is a brief explanation of each field:
+- `version` - Should follow the [version constraints](/docs/templates/hcl_templates/blocks/packer#version-constraints).
+- `source` - Should have the plugin's organizational namespace on GitHub and its short name. Packer will be responsible
+to write the full GitHub address.
+  For example, if the source is `sylviamoss/comment`, Packer will download the binaries from `github.com/sylviamoss/packer-plugin-comment`.
+  To learn more about the source field, check out the [Source Address](/docs/plugins#source-addresses) documentation.
+- `comment`- Can be replaced with whatever you want, and the new value will become the name of the plugin.
+  For example:
+    ```hcl
+    packer {
+        required_plugins {
+            bubbled_up = {
+                # ...
+            }
+        }
+    }
+
+    # ...
+
+    build {
+        # ...
+
+        provisioner "bubbled_up" {
+            # ...
+        }
+    }
+    ```
+
+Once the template is upgraded, you can run `packer init example.pkr.hcl` and the output should tell you the installed
+plugin, and the place where it was installed.
+
+```shell
+➜ packerdev init example.pkr.hcl
+Installed plugin sylviamoss/comment v0.2.23 in "/Users/<user>/.packer.d/plugins/github.com/sylviamoss/comment/packer-plugin-comment_v0.2.23_x5.0_darwin_amd64"
+```
+
+`packer init` will only download plugins binary and never delete existing ones. If all the required_plugins are installed,
```

Comment:
```suggestion
`packer init` will only download plugin binaries and never delete existing ones. If all the required_plugins are installed,
```

Reply:
I never know how to write this which one should be plural 🤦🏼